core/domain/suggestion_registry_test.py

```diff
@@ -114,6 +161,362 @@ def test_get_score_part_helper_methods(self):
+            expected_suggestion_dict['target_version_at_submission'],
+            expected_suggestion_dict['status'], self.author_id,
+            self.reviewer_id, expected_suggestion_dict['change'],
+            expected_suggestion_dict['score_category'], self.fake_date)
+
+        suggestion.change.cmd = 'invalid_cmd'
+        with self.assertRaisesRegexp(
+            Exception, 'Expected cmd to be edit_state_property'):
+            suggestion.validate()
+
+    def test_validate_change_property_name(self):
+        expected_suggestion_dict = self.suggestion_dict
+        suggestion = suggestion_registry.SuggestionEditStateContent(
+            expected_suggestion_dict['suggestion_id'],
+            expected_suggestion_dict['target_id'],
+            expected_suggestion_dict['target_version_at_submission'],
+            expected_suggestion_dict['status'], self.author_id,
+            self.reviewer_id, expected_suggestion_dict['change'],
+            expected_suggestion_dict['score_category'], self.fake_date)
+
+        suggestion.change.property_name = 'invalid_property'
+        with self.assertRaisesRegexp(
+            Exception, 'Expected property_name to be content'):
+            suggestion.validate()
+
+    def test_pre_accept_validate_state_name(self):
+        self.save_new_default_exploration('exp1', self.author_id)
+        expected_suggestion_dict = self.suggestion_dict
+        suggestion = suggestion_registry.SuggestionEditStateContent(
+            expected_suggestion_dict['suggestion_id'],
+            expected_suggestion_dict['target_id'],
+            expected_suggestion_dict['target_version_at_submission'],
+            expected_suggestion_dict['status'], self.author_id,
+            self.reviewer_id, expected_suggestion_dict['change'],
+            expected_suggestion_dict['score_category'], self.fake_date)
+
+        suggestion.change.state_name = 'invalid_state_name'
+        with self.assertRaisesRegexp(
+            Exception, 'Expected invalid_state_name to be a valid state name'):
+            suggestion.pre_accept_validate()
+
+    def test_populate_old_value_of_change_with_invalid_state(self):
+        self.save_new_default_exploration('exp1', self.author_id)
+        expected_suggestion_dict = self.suggestion_dict
+        suggestion = suggestion_registry.SuggestionEditStateContent(
+            expected_suggestion_dict['suggestion_id'],
+            expected_suggestion_dict['target_id'],
+            expected_suggestion_dict['target_version_at_submission'],
+            expected_suggestion_dict['status'], self.author_id,
+            self.reviewer_id, expected_suggestion_dict['change'],
+            expected_suggestion_dict['score_category'], self.fake_date)
+
+        suggestion.change.state_name = 'invalid_state_name'
+
+        self.assertIsNone(suggestion.change.old_value)
+
+        suggestion.populate_old_value_of_change()
+
+        self.assertIsNone(suggestion.change.old_value)
```

Comment:
This should be assertIsNotNone right? That's what is expected after suggestion.populate_old_value_of_change(). If it passes like this something is wrong.

EDIT: I just noticed this is with an invalid state name. Can we raise an error here if so?  With None, it doesn't sound right (because you called a function to populate it and it is still None). Good catch btw!

Reply:
Done